core/src/main/java/com/google/zxing/qrcode/encoder/MatrixUtil.java

```diff
@@ -271,12 +271,7 @@ static void embedDataBits(BitArray dataBits, int maskPattern, ByteMatrix matrix)
   // - findMSBSet(1) => 1
   // - findMSBSet(255) => 8
   static int findMSBSet(int value) {
-    int numDigits = 0;
-    while (value != 0) {
-      value >>>= 1;
-      ++numDigits;
-    }
-    return numDigits;
+    return 32 - Integer.numberOfLeadingZeros(value);
```

Comment:
Nope, that would return the 1 bit in that position instead of the number of bit places. For example:

    0b000010 -> 0b000010
    0b000101 -> 0b000100
    0b001111 -> 0b001000

(My apologies for missing that half of your comment the first time.)